tests/ext/test_precheckoutqueryhandler.py

```diff
@@ -103,6 +107,23 @@ async def callback(self, update, context):
             and isinstance(update.pre_checkout_query, PreCheckoutQuery)
         )
 
+    def test_with_pattern(self, pre_checkout_query):
+        handler = PreCheckoutQueryHandler(self.callback, pattern=".*voice.*")
+
+        assert handler.check_update(pre_checkout_query)
+
+        pre_checkout_query.pre_checkout_query.invoice_payload = "nothing here"
+        assert not handler.check_update(pre_checkout_query)
+
+    def test_with_compiled_pattern(self, pre_checkout_query):
```

Comment:
I created a new test for compiled pattern. If that's not desired, I can rewrite it in the above test.

Reply:
It does the job and I'm happy to keep it as is. adding a note here just in case you're interested in pytests capabilities: One can unify the two tests with `pytest.mark.parametrize`.